### PR TITLE
Remove 4.2 deprecated features

### PR DESCRIPTION
**Pull request by youknowriad**
A lot of cleanup in this release, which is nice:

 - The webpack config is simpler
 - The data module is simpler

**Todo**

 - [x] Update CHANGELOG
 - [x] Remove useless tests
 - [x] Remove useless package dependencies.

### CODE REVIEW

File: packages/editor/src/utils/media-upload/media-upload.js [hunk 1]
```diff
@@ -169,11 +169,7 @@ export function mediaUpload( {
 					caption: get( savedMedia, [ 'caption', 'raw' ], '' ),
 					title: savedMedia.title.raw,
 					url: savedMedia.source_url,
-					mediaDetails: {},
```
Comment (youknowriad) [inline packages/editor/src/utils/media-upload/media-upload.js]: Is this the right change @jorgefilipecosta ?

**Review by aduth — DISMISSED** [review]
Loving the red. ❤️

File: lib/client-assets.php [hunk 2]
```diff
@@ -293,7 +293,6 @@ function gutenberg_register_scripts_and_styles() {
 		array(
```
Comment (youknowriad) [inline lib/client-assets.php]: @atimmer Some changes here to keep track of, for 4.2 merge to Core.

File: packages/components/src/tab-panel/index.js [hunk 3]
```diff
@@ -99,7 +91,7 @@ class TabPanel extends Component {
 						className="components-tab-panel__tab-content"
 						tabIndex="0"
 					>
-						{ this.props.children( Object.assign( new String( selectedTab.name ), selectedTab ) ) }
```
Comment (youknowriad) [inline packages/components/src/tab-panel/index.js]: Sad to see this magic trick go :)

**Comment by aduth** [comment]
There appears to be a mobile test failure? cc @hypest

**Comment by hypest** [comment]
👋 , thanks for the ping @aduth . Yeah, the mobile test fails because `setUnknownTypeHandlerName()` [is not found](https://travis-ci.org/WordPress/gutenberg/jobs/445142989#L729). I see that this PR deprecates that function so, good catch from the mobile tests I'd say :).

This should be easy to fix AFAICT. @youknowriad or @aduth , interested in filing a PR to the mobile repo to fix this? A good chance to get a PR merged to the mobile project with your name on it ;)
Let me know if you'd like me to wrangle this instead. In any case, this PR will need to update the mobile submodule ref to point to a `master` with the fix merged.

**Review by aduth — APPROVED** [review]
Needs rebase, but otherwise 👍

**Pull request merged**